src/shadow-embed.js

```diff
@@ -58,16 +59,29 @@ export function createShadowRoot(hostElement) {
     return existingRoot;
   }
 
+  // Define our shadowRoot object to be returned
+  let shadowRoot;
+
   // Native support.
   const shadowDomSupported = getShadowDomSupportedVersion();
   if (shadowDomSupported == ShadowDomVersion.V1) {
-    return hostElement.attachShadow({mode: 'open'});
+    shadowRoot = hostElement.attachShadow({mode: 'open'});
   } else if (shadowDomSupported == ShadowDomVersion.V0) {
-    return hostElement.createShadowRoot();
+    shadowRoot = hostElement.createShadowRoot();
+  } else {
+    // Polyfill.
+    shadowRoot = createShadowRootPolyfill(hostElement);
   }
 
-  // Polyfill.
-  return createShadowRootPolyfill(hostElement);
+  // Ensure that Shadow CSS is supported, and add an id if not
+  const randomId = Math.floor(Math.random() * 10000);
+  const rootId = `i-amphtml-sd-${randomId}`;
+  if (!isShadowDomSupported() || !isShadowCssSupported(shadowRoot)) {
```

Comment:
Since we are here, let's go ahead and always set id and host class - no need for the `if`.